app/controllers/trips_controller.rb

```diff
@@ -14,6 +15,7 @@ def index
 
   def end_trip
     current_user.idle!
+    NotificationsService.notify_users_on_trip(current_user.trips.last)
```

Comment:
You have an `active_trip` helper on `current_user`, you can take it before `idle!`ing the current user.